src/BlockTemplatesController.php

```diff
@@ -53,10 +53,37 @@ public function add_block_templates( $query_result, $query, $template_type ) {
 			return $query_result;
 		}
 
+		$post_type      = isset( $query['post_type'] ) ? $query['post_type'] : '';
 		$template_files = $this->get_block_templates();
 
+		// @todo: Add apply_filters to _gutenberg_get_template_files() in Gutenberg to prevent duplication of logic.
```

Comment:
Agreed that ideally, we'd want to prevent code duplication here. That said, the fix here unblocks us now which is important. We should keep it on our list to revisit eventually though.